src/push_context.rs

```diff
@@ -0,0 +1,397 @@
+        let local_git_root = maybe_git_root.wrap_err("Could not determine current `git_root`. Pass `--git-root` or set `FLAKEHUB_PUSH_GIT_ROOT`, or run `flakehub-push` with the git root as the current working directory")?;
+
+        let local_git_root = local_git_root
+            .canonicalize()
+            .wrap_err("Failed to canonicalize `--git-root` argument")?;
+        let local_rev_info = RevisionInfo::from_git_root(&local_git_root)?;
+
+        // "cli" and "git_ctx" are the user/env supplied info, augmented with data we might have fetched from github/gitlab apis
+
+        let (token, git_ctx) = match (exec_env.clone(), &cli.jwt_issuer_uri) {
+            (ExecutionEnvironment::GitHub, None) => {
+                // GITHUB CI
+                let github_token = cli
+                    .github_token
+                    .0
+                    .clone()
+                    .expect("failed to get github token when running in GitHub Actions");
+
+                let github_graphql_data_result = GithubGraphqlDataQuery::get(
+                    &client,
+                    &github_token,
+                    &project_owner,
+                    &project_name,
+                    &local_rev_info.revision,
+                )
+                .await?;
+
+                let git_ctx = GitContext::from_cli_and_github(cli, &github_graphql_data_result)?;
+
+                let token = crate::github::get_actions_id_bearer_token(&cli.host)
+                    .await
+                    .wrap_err("Getting upload bearer token from GitHub")?;
+
+                (token, git_ctx)
+            }
+            (ExecutionEnvironment::GitLab, None) => {
+                // GITLAB CI
+                let token = crate::gitlab::get_runner_bearer_token()
+                    .await
+                    .wrap_err("Getting upload bearer token from GitLab")?;
+
+                let git_ctx = GitContext::from_cli_and_gitlab(cli, local_rev_info).await?;
+
+                (token, git_ctx)
+            }
+            (ExecutionEnvironment::Local, Some(u)) => {
+                // LOCAL, DEV (aka emulating GITHUB)
+                let github_token = cli
+                    .github_token
+                    .0
+                    .clone()
+                    .expect("failed to get github token when running locally");
+
+                let github_graphql_data_result = GithubGraphqlDataQuery::get(
+                    &client,
+                    &github_token,
+                    &project_owner,
+                    &project_name,
+                    &local_rev_info.revision,
+                )
+                .await?;
+
+                let git_ctx: GitContext =
+                    GitContext::from_cli_and_github(cli, &github_graphql_data_result)?;
+
+                let token = flakehub_auth_fake::get_fake_bearer_token(
+                    u,
+                    &project_owner,
+                    &format!("{}/{}", project_owner, project_name),
```

Comment:
We already have this in the form of the `repository` variable from up above (~line 86), can we use that instead of reconstructing it?